File: frontend/src/app/services/empleados/empleados.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { BaseResponse } from "src/app/models/base-response";
import { Empleado } from "src/app/models/empleado";
import { Pagination } from "src/app/models/pagination";
import { environment } from "src/environments/environment";

@Injectable({
    providedIn: 'root'
  })
  export class EmpleadosService {
  
    constructor(
      private http: HttpClient,
    ) { }
  
    getAll({page = 1, limit = 10, search}: {page: number, limit: number, search?: string}) {

        const params: any = {
            page: page.toString(),
            limit: limit.toString(),
        };

        if (search) {
            params['search'] = search;
        }

        return this.http.get<BaseResponse<Pagination<Empleado>>>(`${environment.apiUrl}/empleados`, {
            params,
        });
    }

}

File: polizas-faltantes/src/main/java/com/coppel/polizasfaltantes/services/EmpleadosService.java
package com.coppel.polizasfaltantes.services;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.coppel.polizasfaltantes.models.Empleado;
import com.coppel.polizasfaltantes.models.Pagination;
import com.coppel.polizasfaltantes.repositories.EmpleadosRepository;

@Service
public class EmpleadosService {

    @Autowired
    private EmpleadosRepository empleadosRepository;

    public Pagination<Empleado> getAll(int page, int limit, String search) {
        return this.empleadosRepository.getAll(page, limit, search);
    }
}
